packages/ui/src/theme/css/component/table.scss

```diff
@@ -5,75 +5,149 @@
+        --amplify-components-table-header-border-width
+      )
+      var(--amplify-components-table-header-border-width)
+      var(--amplify-components-table-header-border-width)
+      var(--amplify-components-table-header-border-width);
+  }
+
+  &__caption {
     caption-side: var(--amplify-components-table-caption-caption-side);
     color: var(--amplify-components-table-caption-color);
     display: var(--amplify-components-table-caption-display);
-    font-size: var(--amplify-components-table-caption-font-size);
+    font-size: var(--amplify-internal-table-caption-font-size);
     text-align: var(--amplify-components-table-caption-text-align);
     word-break: var(--amplify-components-table-caption-word-break);
   }
 
-  .amplify-table__head {
+  &__head {
     display: var(--amplify-components-table-head-display);
     vertical-align: var(--amplify-components-table-head-vertical-align);
   }
 
-  .amplify-table__body {
+  &__body {
     display: var(--amplify-components-table-body-display);
     vertical-align: var(--amplify-components-table-body-vertical-align);
   }
 
-  .amplify-table__foot {
+  &__foot {
     display: var(--amplify-components-table-foot-display);
     vertical-align: var(--amplify-components-table-foot-vertical-align);
   }
 
-  .amplify-table__row {
+  &__row {
     display: var(--amplify-components-table-row-display);
     vertical-align: var(--amplify-components-table-row-vertical-align);
   }
 
-  .amplify-table__th {
+  &__th {
     border-color: var(--amplify-components-table-header-border-color);
     border-style: var(--amplify-components-table-header-border-style);
-    border-width: var(--amplify-components-table-header-border-width);
+    border-width: var(--amplify-internal-table-th-border-width);
     color: var(--amplify-components-table-header-color);
     display: var(--amplify-components-table-header-display);
-    font-size: var(--amplify-components-table-header-font-size);
+    font-size: var(--amplify-internal-table-th-font-size);
     font-weight: var(--amplify-components-table-header-font-weight);
-    padding: var(--amplify-components-table-header-padding);
+    padding: var(--amplify-internal-table-th-padding);
     vertical-align: var(--amplify-components-table-header-vertical-align);
+    &:first-child {
+      border-left-width: var(--amplify-components-table-header-border-width);
+    }
+    &:last-child {
+      border-right-width: var(--amplify-components-table-header-border-width);
+    }
   }
 
-  .amplify-table__td {
+  &__td {
     border-color: var(--amplify-components-table-data-border-color);
     border-style: var(--amplify-components-table-data-border-style);
-    border-width: var(--amplify-components-table-data-border-width);
+    border-width: var(--amplify-internal-table-td-border-width);
     color: var(--amplify-components-table-data-color);
     display: var(--amplify-components-table-data-display);
-    font-size: var(--amplify-components-table-data-font-size);
```

Comment:
Isn't this an breaking change to remove a previously exposed CSS variable and mark it as internal?

Reply:
That is a really good callout, We should be ok here because this isn't just removed it is instead being set to the new `--amplify-internal-table-td-font-size` which is being set to `--amplify-components-table-data-font-size` on line 8 of this file. But this warrants more investigation to ensure that all of the previously available theming experiences are still available with this new css layout.